includes/shortcodes/class.llms.shortcode.checkout.php

```diff
@@ -298,6 +392,165 @@ protected static function setup_plan_and_form_atts( $plan_id, $atts ) {
 		return $atts;
 	}
 
+	/**
+	 * Verifies that the checkout form can be displayed and prints error notices if not.
+	 *
+	 * @since [version]
+	 *
+	 * @param LLMS_Access_Plan $plan    The access plan object.
+	 * @param LLMS_Product     $product The product object.
+	 * @param int              $user_id The user ID.
+	 * @return bool True if checkout form can be displayed, else false.
+	 */
+	public static function verify_checkout_form_is_displayable( $plan, $product, $user_id ) {
+
+		// Ensure the user isn't already enrolled in the product being purchased.
+		if ( ! self::verify_user_is_not_enrolled( $product, $user_id ) ) {
+			return false;
+		}
+
+		// Make sure the access plan's product exists.
+		if ( ! self::verify_product_exists( $product ) ) {
+			return false;
+		}
+
+		// If there are membership restrictions, check that the user is in at least one membership.
+		// This is to combat CHEATERS.
+		if ( ! self::verify_plan_is_available_to_user( $plan ) ) {
+			return false;
+		}
+
+		// Is the course restricted because enrollment capacity has been reached or we're not in the enrollment period?
+		if ( 'course' === $product->get( 'type' ) ) {
+
+			$course = new LLMS_Course( $product->get( 'id' ) );
+
+			if ( ! self::verify_course_enrollment_has_started( $course ) ) {
+				return false;
+			}
+			if ( ! self::verify_course_enrollment_has_not_ended( $course ) ) {
+				return false;
+			}
+			if ( ! self::verify_course_enrollment_has_capacity( $course ) ) {
+				return false;
+			}
+		}
+
+		return true;
+	}
+
+	/**
+	 * Returns true if the course enrollment has capacity,
+	 * else prints an error notice and returns false.
+	 *
+	 * @since [version]
+	 *
+	 * @param LLMS_Course $course The course object.
+	 * @return bool
+	 */
+	public static function verify_course_enrollment_has_capacity( $course ) {
+
+		if ( ! $course->has_capacity() ) {
+			llms_print_notice( $course->get( 'capacity_message' ), 'error' );
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Returns true if the course enrollment end date has not passed or the course does not have an enrollment period,
+	 * else prints an error notice and returns false.
+	 *
+	 * @since [version]
+	 *
+	 * @param LLMS_Course $course The course object.
+	 * @return bool
+	 */
+	public static function verify_course_enrollment_has_not_ended( $course ) {
+
+		if ( $course->has_enrollment_period_ended() ) {
+			llms_print_notice( $course->get( 'enrollment_closed_message' ), 'error' );
```

Comment:
This kind of message contains a shortcode, e.g. 
`Enrollment in this course closed on [lifterlms_course_info id="123" key="enrollment_end_date"].`
So shortcodes need to be parsed.
Basically if you use the pricing table shortcode from outside a course, e.g. in a page, shortcodes inside that message are not parsed, same thing when you display that notice on the checkout page (at least from my tests).
The reason why the shortcodes in the notices are parsed when from within a course content is because:
1) (if the pricing table is added via a shortcode in a course content) see `do_shortcode()` here https://github.com/gocodebox/lifterlms/blob/6.6.0/includes/functions/llms-functions-content.php#L59
2) (if the pricing table is added via a block) [`do_blocks` (priority 9)](https://github.com/WordPress/WordPress/blob/6.0/wp-includes/default-filters.php#L184) applied before `[do_shortcodes` (priority 11)](https://github.com/WordPress/WordPress/blob/6.0/wp-includes/default-filters.php#L603)